tools/piranha/piranha.xml

```diff
@@ -0,0 +1,169 @@
+            -a $background_thresh
+        #end if
+        #if $bin_size_reponse:
+            -b $bin_size_reponse
+        #end if
+        #if $bin_size_covars:
+            -i $bin_size_covars
+        #end if
+        #if $bin_size_both:
+            -z $bin_size_both
+        #end if
+        #if $merge_bins.merge=="yes":
+            -u $merge_bins.cluster_dist
+        #else:
+            -u $merge_bins.merge
+        #end if
+        $suppress_covars
+        $fit
+        #if $dist:
+            -d $dist
+        #end if
+        $fitMethod
+        #if $use_model.model_select=="yes":
+            -m $use_model.model
+        #end if
+        $unstranded
+        $no_normalisation
+        $log_covars
+        ./foo.$input.ext
+        #for $covariate in $covariates:
+            $covariate.covariate_file
+        #end for
+
+        -o ./piranha.out
+    ]]></command>
+    <inputs>
+        <param name="input" type="data" format="bed,bam" label="Choose input BED or BAM file"/>
+        <repeat name="covariates" title="Choose covariate">
+            <param name="covariate_file" type="data" format="bed" label="Choose covariate BED file"/>
+        </repeat>
+        <param name="sort" type="boolean" label="Is your input file already sorted?" checked="False" falsevalue="-s" truevalue=""/>
+        <param name="p_threshold" type="float" min="0" max="1" value="1" label="Significance threshold for sites"/>
+        <param name="no_pval_correct" type="boolean" checked="False" falsevalue="" truevalue="-c"
+            label="Disable p-values correction for multiple hypothesis testing?" help="We correct by default using B &amp; H."/>
+        <param name="background_thresh" type="float" optional="true"
+            label="Indicates that this proportion of the lowest scores should be considered the background"/>
+        <param name="bin_size_reponse" type="integer" optional="true"
+            label="Indicates that input is raw reads and should be binned into bins of this size"/>
+        <param name="bin_size_covars" type="integer" optional="true"
+            label="Indicates that the covariates are raw reads and should be binned into bins of this size"/>
+        <param name="bin_size_both" type="integer" optional="true" label="Synonymous with -b x -i x for any x"/>
+        <conditional name="merge_bins">
+            <param name="merge" type="select" label="Merge significant bins within certain distance?">
+                <option value="yes">Yes</option>
+                <option value="0" selected="true">No</option>
+            </param>
+            <when value="yes">
+                <param name="cluster_dist" type="integer" value="1" label="Merge significant bins within this distance"
+                help="Default is 1 (merge adjacent)"/>
+            </when>
+        </conditional>
+        <param name="suppress_covars" type="boolean" checked="False" truevalue="-r" falsevalue="" label="Don't print covariate values in output" />
```

Comment:
I usually find the positive formulation easier to comprehend:
so: label="Print covariate values in output" checked="True" truevalue="" falsevalue="-r"